var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -37,6 +37,7 @@ class Upcxx(Package):
               msg='None is unacceptable on Cray.')
 
     depends_on('cuda', when='+cuda')
+    depends_on('python@2.7.5:2.999')
```

Comment:
Python is usually a `type=('build', 'run')` dependency, how is it used by the UPC++ package?

Reply:
> Also, does UPC++ have any plans for Python 3 support in the future? Python 2 is officially EOL.

The current version requires Python2 2.7.[5+] for package build/install and 2.7.[5+] or 3.0+ for package use - ie the job launch wrapper script used to launch applications is already Python3 compatible. So I believe @adamjstewart is correct we need to add `type=('build', 'run')` to override the default of `('build', 'link')`

The next version (expected in March) will drop the Python build/install requirements entirely and merely require 2.7.[5+] or 3.0+ for package use - but we will update Spack to reflect that when the new version is available (which will probably be something like `depends_on('python@2.7.5:',type=('run'))` )

Reply:
Just updated the PR to reflect the fact that current python dependency is build+run. Also added @bonachea as maintainer. @adamjstewart